### DISCUSSION TITLE
Deegram for Bubble.io - having trouble getting recording started

### DISCUSSION THREAD

**Discussion by DariusNmn** [deepgram/community#182] [General help] [answered]
Hello, I'm having a few issues getting Deepgram to start recording on Bubble.io.

From what I understand the initial workflow seems very simple and just involves using the startRecording command, and setting a text box to Deepgram's Transcription. This is what I've done, but the text box only says "starting..." and the indicator I've setup to tell me whether or not recording has started doesn't indicate that it has started.

My current setup:
1. If button clicked > startRecording DeepgramRecorder A.
2. Text box contents set to: DeepgramRecoder A's Transcription.

Any input would be greatly appreciated.

**Comment by jjmaldonis** [comment]
Hey @DariusNmn! I haven't used Bubble.io so some additional info would be helpful. Are you using [this plugin](https://bubble.io/plugin/deepgram-realtime-transcription-1676726070867x623253573730041900)? If not, how are you writing the integration with Deepgram?

I'm not aware of the buttons or the `startRecording` command, so if you have screenshots and URLs - or anything else - that would be very helpful.

**Comment by DariusNmn** [comment]
Hello, thank you for the speedy response! I also appreciate your willingness to help.

Yes, you are correct. I was using the plugin that you linked.
[Here](https://bubble.io/page?type=page&name=index&id=deepgram-plugin&tab=tabs-1) is the example application the plugin developer provides.

All I am trying to do it start realtime transcription and have the text arrive in realtime into a text box, but I have made little progress.

These are the element actions provided by the plugin: 
![image](https://github.com/deepgram/community/assets/54332287/a9136d14-80f5-4b3b-8c20-9403b6a4bbfd)

These are the plugin actions provided by the plugin:
![image](https://github.com/deepgram/community/assets/54332287/84c19f4c-7b25-4c97-8535-3a7c306ce23a)

The initial workflow appears to be extremely simple in the provided example application I linked, but it doesn't seem to work in practice: 
![image](https://github.com/deepgram/community/assets/54332287/de08b146-2e81-4f11-86c7-030a741913f2)

It doesn't even ask for microphone permissions in testing in the deployed app. The text field, which is set to the value of "DeepgramRecorder A's transcription" says "Starting..." when I start the workflow, but it's clear that no recording ever begins

Is it possible to do the same kind of realtime transcription through bubble.io with API calls as it is with the plugin? I have read through quite a bit of the documentation but I wasn't able to find something which confirmed whether or not realtime transcription was possible without the plugin. It would be ideal to get it working with the plugin, but for the life of me I cannot figure out why it isn't functioning, and the plugin documentation is extremely limited.

Any advice would be greatly appreciated. Thank you again for getting back to me initially.

**Reply by wapdat** [reply, marked as answer]
Hi @DariusNmn  - I'm Lindsay - the author of the Bubble Plugin.
Deepgram support @jjmaldonis got in touch with me and let me know about your issue. It's my plugin problem - not Deepgram here.

Either DM me in the Bubble forum @lindsay_knowcode or email me at lindsay@knowcode.tech

You can see a simplified Bubble Editor setup here of how the Plugin works
https://bubble.io/page?type=page&name=index&id=deepgram-plugin&tab=tabs-1

You can see the documentation for the plugin here 
https://planbbackups.io/deepgram

My guess will be that you haven't set up the Backend workflows necessary to get temporary access tokens to Deepgram. The reasons for this are explained in the link above and there is a troubleshooting section that I think explains your symptoms, cause and fix.

If you still have not got it working after this - if you are happy to make me a collaborator on our app I can log in, inspect your Bubble app, and fix it.  (No charge - purchase of the plugin comes with reasonable support to get you up and running)

I don't monitor this github forum so best to email me. 

Cheers
Lindsay
